### ISSUE TITLE
Add support for static and dynamic quantization with Conv1D operator

### ISSUE THREAD

**Issue by raghuramank100** [pytorch/pytorch#34074]
Conv1D is invoked as part of GPT2 (instead of matmul) and is also used in speech models. We will need dynamic quant support and static quant support for this operator in pytorch. Also, see: 
https://github.com/pytorch/pytorch/issues/33433
https://github.com/pytorch/pytorch/issues/33487
#32768 

cc @jerryzh168 @jianyuh @dzhulgakov @raghuramank100 @jamesr66a

**Comment by supriyar** [comment]
Static quantization support for conv1D is done.
cc @dskhudia to check the feasibility of implementing server side kernel for dynamic quant for this op.

**Comment by Aravindbolloju** [comment]
Hello Sir..
@raghuramank100 @supriyar @dskhudia 
I build a pytorch model based on conv1d. I gone through quantization and implemented some cases as well but all those are working on conv2d, bn,relu but In my case, my model is built on conv1d and PReLU. Does this quatization valid for these network layers? Because when I did quantization only the layers which are included in mapping is only quantized. Let me show you those layers for which quantization is valid(i.e which are included in mapping)
Please find the list of modules it supports:(according to source code in went through)
  (Actual layer : quantized layer)
    nn.Linear: nnq.Linear,
    nn.ReLU: nnq.ReLU,
    nn.ReLU6: nnq.ReLU6,
    nn.Conv2d: nnq.Conv2d,
    nn.Conv3d: nnq.Conv3d,
    nn.BatchNorm2d: nnq.BatchNorm2d,
    nn.BatchNorm3d: nnq.BatchNorm3d,
    QuantStub: nnq.Quantize,
    DeQuantStub: nnq.DeQuantize,
    # Wrapper Modules:
    nnq.FloatFunctional: nnq.QFunctional,
    # Intrinsic modules:
    nni.ConvReLU2d: nniq.ConvReLU2d,
    nni.ConvReLU3d: nniq.ConvReLU3d,
    nni.LinearReLU: nniq.LinearReLU,
    nniqat.ConvReLU2d: nniq.ConvReLU2d,
    nniqat.LinearReLU: nniq.LinearReLU,
    nniqat.ConvBn2d: nnq.Conv2d,
    nniqat.ConvBnReLU2d: nniq.ConvReLU2d,
    # QAT modules:
    nnqat.Linear: nnq.Linear,
    nnqat.Conv2d: nnq.Conv2d,

Is it, it means that quantization can't be done on conv1d and PReLU?

Thankyou , Sir.

**Comment by snakers4** [comment]
Hi @dskhudia, 

Are you still planning to release this?

Checked in releases and in the [docs](https://pytorch.org/docs/stable/quantization.html#torch-nn-intrinsic), looks like it was not released yet

**Comment by dskhudia** [comment]
Server side conv1d addition is still in progress. cc @shz0116

**Comment by supriyar** [comment]
looks like quantized_conv1d support was added for both server (fbgemm) and mobile (qnnpack). Closing this issue
